packages/flutter_tools/test/integration.shard/hot_reload_test.dart

```diff
@@ -16,65 +17,100 @@ import 'test_utils.dart';
-  final HotReloadProject _project = HotReloadProject();
-  FlutterRunTestDriver _flutter;
+  final HotReloadProject project = HotReloadProject();
+  FlutterRunTestDriver flutter;
 
   setUp(() async {
     tempDir = createResolvedTempDirectorySync('hot_reload_test.');
-    await _project.setUpIn(tempDir);
-    _flutter = FlutterRunTestDriver(tempDir);
+    await project.setUpIn(tempDir);
+    flutter = FlutterRunTestDriver(tempDir);
   });
 
   tearDown(() async {
-    await _flutter?.stop();
+    await flutter?.stop();
     tryToDelete(tempDir);
   });
 
   test('hot reload works without error', () async {
-    await _flutter.run();
-    await _flutter.hotReload();
+    await flutter.run();
+    await flutter.hotReload();
   });
 
   test('newly added code executes during hot reload', () async {
     final StringBuffer stdout = StringBuffer();
-    final StreamSubscription<String> subscription = _flutter.stdout.listen(stdout.writeln);
-    await _flutter.run();
-    _project.uncommentHotReloadPrint();
+    final StreamSubscription<String> subscription = flutter.stdout.listen(stdout.writeln);
+    await flutter.run();
+    project.uncommentHotReloadPrint();
     try {
-      await _flutter.hotReload();
+      await flutter.hotReload();
       expect(stdout.toString(), contains('(((((RELOAD WORKED)))))'));
     } finally {
       await subscription.cancel();
     }
   });
 
-  test('reloadMethod triggers hot reload behavior', () async {
+  test('fastReassemble behavior triggers hot reload behavior with evaluation of expression', () async {
     final StringBuffer stdout = StringBuffer();
-    final StreamSubscription<String> subscription = _flutter.stdout.listen(stdout.writeln);
-    await _flutter.run();
-    _project.uncommentHotReloadPrint();
+    final StreamSubscription<String> subscription = flutter.stdout.listen(stdout.writeln);
+    await flutter.run(withDebugger: true);
+
+    final int port = flutter.vmServicePort;
+    final VmService vmService = await vmServiceConnectUri('ws://localhost:$port/ws');
     try {
-      final String libraryId = _project.buildBreakpointUri.toString();
-      await _flutter.reloadMethod(libraryId: libraryId, classId: 'MyApp');
-      // reloadMethod does not wait for the next frame, to allow scheduling a new
-      // update while the previous update was pending.
-      await Future<void>.delayed(const Duration(seconds: 1));
-      expect(stdout.toString(), contains('(((((RELOAD WORKED)))))'));
+      // Since the single-widget reload feature is not yet implemented, manually
+      // evaluate the expression for the reload.
+      final Isolate isolate = await waitForExtension(vmService);
+      final LibraryRef targetRef = isolate.libraries.firstWhere((LibraryRef libraryRef) {
+        return libraryRef.uri == 'package:test/main.dart';
+      });
+      await vmService.evaluate(
+        isolate.id,
+        targetRef.id,
+        '((){debugFastReassembleMethod=(Object _) => _ is MyApp})()',
```

Comment:
Isn't `_` reserved for "ignored" params per style giude?

Reply:
I'm being overly defensive because there is a chance that a user creates a widget which matches the parameter name, and `_` seems like the least-likely collision

Reply:
I could name it like _param_flutter_please_dont_collide_123223232 - that seems less likely

Reply:
For the test it doesn't really matter, so I named it `x`